pkg/asset/installconfig/installconfig.go

```diff
@@ -192,9 +191,6 @@ func (a *InstallConfig) platformValidation() error {
 	if a.Config.Platform.Ovirt != nil {
 		return icovirt.Validate(a.Config)
 	}
-	if a.Config.Platform.OpenStack != nil {
-		return icopenstack.Validate(a.Config)
-	}
```

Comment:
I don't think you want to remove this. Install config validation should be done when creating assets like manifests or ignition configs. Try `create manifests` with this change and see if your validation is catching.

Reply:
I agree that logically the validation should be here, but it returns incorrect results because the asset will be modified later.
Therefore I see two solutions here:

1. Move the validation to "cluster" asset where it works properly (this PR).
2. Stop modifying install-config in different places.